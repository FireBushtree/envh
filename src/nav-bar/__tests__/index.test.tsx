import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NavBar from '../nav-bar';

describe('<NavBar />', () => {
  test('should render default', () => {
    const component = render(<NavBar>nav-bar</NavBar>);
    const { container } = component;
    expect(container).toMatchSnapshot();
  });

  test('render left, right by props', () => {
    render(
      <NavBar left="left" right="right">
        nav-bar
      </NavBar>,
    );

    expect(screen.getByText('left')).toBeTruthy();
    expect(screen.getByText('right')).toBeTruthy();
  });

  test('click', () => {
    let leftCount = 0;
    let rightCount = 0;

    render(
      <NavBar
        left="left"
        right="right"
        onLeftClick={() => {
          leftCount += 1;
        }}
        onRightClick={() => {
          rightCount += 1;
        }}
      >
        nav-bar
      </NavBar>,
    );

    userEvent.click(screen.getByText('left'));
    expect(leftCount).toBe(1);
    userEvent.click(screen.getByText('right'));
    expect(rightCount).toBe(1);

    userEvent.click(screen.getByText('left'));
    expect(leftCount).toBe(2);
  });
});
