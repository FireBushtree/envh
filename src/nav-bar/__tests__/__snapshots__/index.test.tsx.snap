// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render default 1`] = `
<div>
  <div
    class="eh-nav-bar"
  >
    <div
      class="eh-nav-bar-action"
    >
      <img
        alt="left"
        class="eh-nav-bar-back"
        src="[object Object]"
      />
    </div>
    <div
      class="eh-nav-bar-content"
    >
      nav-bar
    </div>
    <div
      class="eh-nav-bar-action"
    />
  </div>
</div>
`;
